### ISSUE TITLE
JavaScript highlighting broken after template string.

### ISSUE THREAD

**Issue by aeschli** [Microsoft/TypeScript-TmLanguage#120]
_From @martinbilles on March 9, 2016 17:7_

Syntax hightlighting for JavaScript breaks if a variable used inside a template string escape sequence starts with the characters "var":

```javascript
var objid, varname;
{
    console.log(`${objid}.${varname}`);
}
thisPartIsWrong();
```

Here, the variable name ```varname``` is incorrectly highlighted, the "var" part is highlighted in the keyword color, while the rest of the variable name is colored correctly. After the template string, all remaining code is colored in the string color.

![vscode-highlighting-bug](https://cloud.githubusercontent.com/assets/5391834/13643515/9a63e41a-e621-11e5-92c9-a050cb167492.png)

This is only true if the variable's name starts with "var" or "let". If any other keyword is used, e.g. "constname", "ifname", there is no error. if there is no wrapping block statement around the code containing the template string, the error also doesn't appear.

_Copied from original issue: Microsoft/vscode#3921_

**Comment by martinbilles** [comment]
This appears fixed in Visual Studio Code 1.0.0.

**Comment by sheetalkamat** [comment]
Fixed by #257
